src/components/TranslationProvider.tsx

```diff
@@ -0,0 +1,25 @@
+import React, { ReactElement, useEffect, useState } from "react";
+import { TranslationContext } from "../contexts/TranslationContext";
+import { LANGS } from "../types/TranslationTypes";
+import defaultLang from "../i18n/en.json";
+
+export default function TranslationProvider(props: { children: ReactElement }) {
+  const [config, changeConfig] = useState(defaultLang);
+  const [currentLang, changeLang] = useState(LANGS.en);
```

Comment:
It's always better when you give more appropriate name for state update callback.
It would make it easier for other developers to work with if you'll name those using such pattern as "set..".

`  const [config, setConfig] = useState(defaultLang);`
  `const [currentLang, setCurrentLang] = useState(LANGS.en);`

Reply:
agree with you, fixed